build/net/src/lib.rs

```diff
@@ -82,23 +90,65 @@ pub fn load_net_config() -> Result<NetConfig> {
     Ok(cfg)
 }
 
+pub fn generate_port_consts(
+    config: &NetConfig,
+    mut out: impl std::io::Write,
+) -> Result<(), std::io::Error> {
+    let ports = config
+        .vlans
+        .iter()
+        .map(|v| v.port)
+        .collect::<HashSet<_>>()
```

Comment:
I had to stare at this for a bit to convince myself that you don't wind up relying on the iteration order of HashSet. (Because using HashSet in the build system in general is a Big Red Flag.)

Very weak preference for BTreeSet here just to avoid looking sus.

If you were open to `cargo add itertools` you could rephrase this as `.unique().count()` (you'd have to `use itertools::Itertools`).

Reply:
I am against bad vibes and looking sus; changed to `BTreeSet` in 1528c24be825